### PR TITLE
New Package: kubernetes-helm

### PR DESCRIPTION
**Pull request by knusbaum**
This PR adds a package for kubernetes-helm, a package-manager for kubernetes applications.

### CODE REVIEW

**Comment by knusbaum** [comment]
Thanks for the feedback! I'll work on these tonight, or in the next couple days.

**Comment by knusbaum** [comment]
I fixed the commit message and some of the comments.
The comment I did not fix was the one about build_style=go.

`build_style=go` does help with the extraction by creating a `$GOPATH`, and symlinking the project into it. Unfortunately, the `go` tool, and some other go-related tools seem to not like symlinks in the `$GOPATH`, and in certain cases will refuse to resolve them. The build system for this project seems to be one that suffers from that issue. If the project is un-tarred at the location, the build works, but if it is a symlink, the build fails to find the subdirectories, and builds nothing.

Another issue with `build_style=go` is that it sets `$GOARCH` and a couple of other variables, which causes issues with the build. The build itself has tooling to do cross-compilation, and expects these variables to be set for the host architecture, otherwise it ends up building and trying to execute some precursor binaries, which are then in the wrong exec format.

I can probably make it work with some effort, but these issues lead me to believe that `build_style=go` is just not a good fit for this package.

File: srcpkgs/kubernetes-helm/template [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+maintainer="Kyle Nusbaum <knusbaum+void@sdf.org>"
+license="Apache-2.0"
+homepage="https://helm.sh/"
+distfiles="https://github.com/helm/helm/archive/v${version}.tar.gz"
+checksum=2fee31764117024d1cc4f597d23791c60feef7c3ceedd143cfdbbafca7a7e126
+nopie=true
+conflicts="helm"
+wrksrc="helm-${version}"
+
+convert_arch() {
```
[the remark this answers is not in the document]
Comment (knusbaum) [reply srcpkgs/kubernetes-helm/template]: Addressed

**Comment by Gottox** [comment]
AFAICS there's no need for defining a buildstyle as you overwrite the do_{build,install} functions and the do_check function is broken with this package anyway.

In the long term it would be cool to use build_style=go with any package that uses the go build system, but for now I'm fine with it.

**Comment by ebfe** [comment]
> _go_target_arch=$(source /void-packages/common/environment/build-style/go.sh; echo $GOARCH)

This is broken on most systems.

**Comment by dan-santana** [comment]
@knusbaum would you mind adding the man pages and the completions for bash/zsh in this package? There are some snippets [in here](https://github.com/voidlinux/void-packages/pull/13200/files) which might come in handy for this.

Thank you!

**Pull request merged**